app/views/posts/show.html.erb

```diff
@@ -6,7 +6,11 @@
     <div class="card mb-5">
       <div class="card-body ">
         <% @post.comments.each do |comment| %>
-        <h5><%=comment.user.name%> : <%=comment.text%> </h5>
+        <h5 class="d-flex"><%=comment.user.name%> : <%=comment.text%>
+         <% if can? :destroy, comment %>
+           <%= button_to 'Delete', user_post_comment_path(@post.author_id, @post,comment), method: :delete, class: 'btn btn-light text-danger btn-sm'%>
+         <% end %>
+       </h5>
```

Comment:
- [ ] Great work on implementing authorization using the `CanCanCan` gem! Your efforts are commendable. However, there's a small enhancement that can elevate your project even further:

- **Authorization for Non-Admin Roles:** Currently, users with roles other than `admin` are unable to interact with posts. Upon attempting to access the post details page to `view`, `comment`, or `like` a post, an error is displayed. I've attached a screenshot illustrating this issue.

  
- 🚀 **Recommendation for Improvement:** To address this, consider refining your authorization logic to ensure that users with roles other than `admin` have the necessary permissions to interact with posts. This will enhance the user experience and functionality of your application.

<img width="1141" alt="Screenshot 2023-12-14 at 19 09 37" src="https://github.com/AlselwiAisha/BlogApp/assets/107192076/5307da37-4f3f-4d4c-abe0-2a06df352d9e">